File: backend/src/main/java/com/lily/backend/user/UserController.java
package com.lily.backend.user;

import com.lily.backend.common.dto.APIResponse;
import com.lily.backend.user.dto.UserDto;
import com.lily.backend.user.request.LoginRequest;
import com.lily.backend.user.request.SignupRequest;
import com.lily.backend.user.response.JwtResponse;
import com.lily.backend.user.response.SingupResult;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


@RestController
@RequestMapping("/api/v1/user")
public class UserController {

  @Autowired
  private UserService userService;

  @PostMapping("/register")
  public APIResponse<SingupResult> register(@RequestBody final SignupRequest request) {

    UserDto newUser = userService.register(request);

    SingupResult result = SingupResult.success(newUser);
    return APIResponse.success(result);
  }

  @PostMapping("/login")
  public APIResponse<JwtResponse> signIn(@RequestBody final LoginRequest loginRequest) {

    JwtResponse jwtResponse = userService.authenticate(loginRequest);

    return APIResponse.success(jwtResponse);
  }
}

File: backend/blog-service/src/main/kotlin/com/lily/blogservice/endpoint/UserController.kt
package com.lily.blogservice.endpoint

import com.lily.blogservice.data.APIResponse
import com.lily.blogservice.data.LoginRequest
import com.lily.blogservice.data.SignupRequest
import com.lily.blogservice.data.SingupResult
import com.lily.blogservice.handler.UserAuthHandler
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.reactive.function.BodyInserters.fromObject
import org.springframework.web.reactive.function.server.ServerResponse

@RestController
@RequestMapping("/api/v1/user")
class UserController(val handle: UserAuthHandler) {

    @PostMapping("/register")
    fun register(@RequestBody request: SignupRequest) = handle.register(request).map { APIResponse(result = SingupResult(user = it), code = 200) }

//    @PostMapping("/login")
//    fun login(@RequestBody login: LoginRequest) = handle.login(login).map { APIResponse(result = it, code = 200) }.flatMap {
//        ServerResponse.ok()
//        .contentType(MediaType.APPLICATION_JSON)
//        .body(fromObject(it))
//        }
}